kale/loaddata/multiomics_gnn_dataset.py

```diff
@@ -0,0 +1,459 @@
+# =============================================================================
+# Author: Sina Tabakhi, stabakhi1@sheffield.ac.uk
+# =============================================================================
+
+"""
+Construct a dataset with multiple omics modalities based on PyTorch Geometric.
+
+This code is written by refactoring MOGONET dataset code (https://github.com/txWang/MOGONET/blob/main/train_test.py)
+within 'Dataset' class provided in the PyTorch Geometric.
+
+Reference:
+Wang, T., Shao, W., Huang, Z., Tang, H., Zhang, J., Ding, Z., Huang, K. (2021). MOGONET integrates multi-omics data
+using graph convolutional networks allowing patient classification and biomarker identification. Nature communications.
+https://www.nature.com/articles/s41467-021-23774-w
+"""
+
+import os.path as osp
+from typing import Callable, List, Optional, Tuple, Union
+
+import numpy as np
+import torch
+import torch.nn.functional as F
+from torch_geometric.data import Data, Dataset, download_url, extract_zip
+from torch_sparse import SparseTensor
+
+from kale.prepdata.distance import calculate_distance, DistanceMetric
+
+
+class MultiOmicsDataset(Dataset):
+    r"""The multiomics data for creating graph dataset.
+
+    Args:
+        root (string): Root directory where the dataset should be saved.
+        num_view (int): The total number of modalities in the dataset.
+        num_class (int): The total number of classes in the dataset.
+        url (string, optional): The url to download the dataset from.
+        raw_file_names (list[callable], optional): The name of the files in the :obj:`self.raw_dir` folder that must be
+            present in order to skip downloading.
+        random_split (bool, optional): Whether to split the dataset into random train and test subsets. (default:
+            :obj:`False`)
```

Comment:
You like to use :obj: etc a lot. Is it necessary? Is it more readable? Is this what PyTorch is doing?